### PR TITLE
Login and Registration

### PR DESCRIPTION
**Pull request by gorg16**

### CODE REVIEW

File: src/app/login/login.component.ts [hunk 1]
```diff
@@ -31,8 +33,33 @@ export class LoginComponent implements OnInit {
     // const {value} = this.loginForm.get('email');
     // console.log(this.authService.getUsers());
     // if (value === this.users[0].userEmail) {
-    this.router.navigate(['/servers']);
-  // }
+    this.storageUsers = localStorage.getItem('Users');
+    if (this.storageUsers) {
+    this.storageUsers = JSON.parse(this.storageUsers);
```
Comment (narek118) [inline src/app/login/login.component.ts]: use tslint, this line should be one tab right,

File: src/app/login/login.component.ts [hunk 2]
```diff
@@ -31,8 +33,33 @@ export class LoginComponent implements OnInit {
...
+    this.currentUser = this.storageUsers.find((r) => {
```
Comment (narek118) [inline src/app/login/login.component.ts]: this.storageUsers.find((user) => {

File: src/app/login/login.component.ts [hunk 3]
```diff
@@ -31,8 +33,33 @@ export class LoginComponent implements OnInit {
...
+    //   console.log(users[user]);
+    //    // if (user) // user === login linox userin
+    //    // {
+    //     // login anel
+    //   this.router.navigate(['/servers']);
+    //
+    //    // }
+    // }
+    // console.log((localStorage.getItem('Users')));
+    if (localStorage.getItem('Users')) {
```
Comment (narek118) [inline src/app/login/login.component.ts]: delete unused codes

File: src/app/register/register.component.html [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 <h2 style="text-align: center">Register</h2>
 <div class="example-form" style="text-align: center">
-  <form [formGroup]="registerForm"  >
+  <form [formGroup]="registerForm" #f >
```
Comment (narek118) [inline src/app/register/register.component.html]: don't use `#f`. Never use short uncertain variables. User #registerForm.

File: src/app/register/register.component.ts [hunk 5]
```diff
@@ -27,8 +28,32 @@ export class RegisterComponent implements OnInit {
 
 
   onRegister() {
+    this.serversService.serverTypes.next(this.registerForm.get('types').value);
     console.log('aaaaaaa');
```
Comment (narek118) [inline src/app/register/register.component.ts]: Remove all console.logs !!!

File: src/app/servers/header/header.component.ts [hunk 6]
```diff
@@ -6,10 +6,27 @@ import { Component, OnInit } from '@angular/core';
...
 })
 export class HeaderComponent implements OnInit {
-
+  usernaeme: any;
+  storageUsers: any;
+  currentUserName: any;
   constructor() { }
 
   ngOnInit() {
+    this.currentUserName = JSON.parse(localStorage.getItem('currentUser')).name;
```
Comment (narek118) [inline src/app/servers/header/header.component.ts:16]: You should have used localStorage service (service to work with localstorage). You should never add `JSON.parse(localStorage.getItem ` stuff in any component but instead use storageService.get('currentUser')

File: src/app/servers/server-detail/server-detail.component.ts [hunk 7]
```diff
@@ -14,28 +14,29 @@ export class ServerDetailComponent implements OnInit {
...
-  animal: string;
-  name: string;
+
   constructor(private serversService: ServersService, private  route: ActivatedRoute, private dialog: MatDialog) { }
 
   ngOnInit() {
     this.servers = this.serversService.getServers();
 
-    console.log('texsz');
+    console.log(this.id, 'texsz');
```
Comment (narek118) [inline src/app/servers/server-detail/server-detail.component.ts]: remove

File: src/app/servers/server-detail/server-detail.component.ts [hunk 8]
```diff
@@ -14,28 +14,29 @@ export class ServerDetailComponent implements OnInit {
...
     // const id = this.route.snapshot.params['id'];
     this.route.params
       .subscribe(
         (parmas: Params) => {
           this.id = +parmas['id'];
+          console.log('aaa');
+
           this.server = this.serversService.getServer(this.id);
+          console.log(this.server, 'saaa');
```
Comment (narek118) [inline src/app/servers/server-detail/server-detail.component.ts:27]: remove

File: src/app/servers/server-detail/server-detail.component.ts [hunk 9]
```diff
@@ -14,28 +14,29 @@ export class ServerDetailComponent implements OnInit {
...
     this.servers = this.serversService.getServers();
 
-    console.log('texsz');
+    console.log(this.id, 'texsz');
     // const id = this.route.snapshot.params['id'];
     this.route.params
       .subscribe(
         (parmas: Params) => {
           this.id = +parmas['id'];
+          console.log('aaa');
```
Comment (narek118) [inline src/app/servers/server-detail/server-detail.component.ts]: remove

File: src/app/servers/server-list/server-item/server-item.component.ts [hunk 10]
```diff
@@ -9,8 +10,39 @@ import {Server} from '../../servers.modal';
...
+  web;
+  // @Input('f') form;
+
+  constructor(private serversService: ServersService) {}
 
   ngOnInit() {
+    // console.log(this.serverType );
+    this.serversService.serverTypes.subscribe( type => {
+    this.serverType = type;
+    console.log(this.index, 'blablabla')
```
Comment (narek118) [inline src/app/servers/server-list/server-item/server-item.component.ts]: remove

File: src/app/servers/server-list/server-item/server-item.component.ts [hunk 11]
```diff
@@ -9,8 +10,39 @@ import {Server} from '../../servers.modal';
...
+    // console.log(this.serverType);
+    // this.ssh = this.serverType ? this.serverType.includes('ssh') : null;
+    // this.http = this.serverType ? this.serverType.includes('http') : null;
+    // this.web = this.serverType ? this.serverType.includes('Web') : null;
+    // console.log( this.http, this.ssh, this.web);
+    });
+
+    // this.storageUsers = localStorage.getItem('Users');
+
+    // this.storageUsers = JSON.parse(this.storageUsers);
```
Comment (narek118) [inline src/app/servers/server-list/server-item/server-item.component.ts]: too much comments

File: src/app/servers/server-list/server-list.component.ts [hunk 12]
```diff
@@ -12,13 +12,30 @@ export class ServerListComponent implements OnInit, OnDestroy {
...
+  types = [];
+  userService: Server[];
 
 
   constructor(private serversService: ServersService ) { }
 
   ngOnInit() {
 
     this.servers = this.serversService.getServers();
+    this.types =  JSON.parse(localStorage.getItem('currentUser')) ? JSON.parse(localStorage.getItem('currentUser')).types : null;
```
Comment (narek118) [inline src/app/servers/server-list/server-list.component.ts:23]: should use a storage service

File: src/app/servers/server-list/server-list.component.ts [hunk 13]
```diff
@@ -12,13 +12,30 @@ export class ServerListComponent implements OnInit, OnDestroy {
...
+
+    console.log(this.types, 'types');
```
Comment (narek118) [inline src/app/servers/server-list/server-list.component.ts]: remove

File: src/app/servers/server-list/server-list.component.ts [hunk 14]
```diff
@@ -12,13 +12,30 @@ export class ServerListComponent implements OnInit, OnDestroy {
...
+    this.userService = this.servers.filter(item => {
+      // console.log(item.serverType, 'items');
```
Comment (narek118) [inline src/app/servers/server-list/server-list.component.ts]: remove

File: src/app/register/register.component.ts (hunk 5 above)
Comment (narek118) [inline src/app/register/register.component.ts]: removee

File: src/app/servers/server-list/server-list.component.ts [hunk 15]
```diff
@@ -12,13 +12,30 @@ export class ServerListComponent implements OnInit, OnDestroy {
...
+      // console.log(this.types.indexOf(item.serverType))
+
+      for (let i = 0; i <= this.types.length ; i++) {
+        if (item.serverType === this.types[i]) {
+            return true;
+        }
+      }
+      // this.types.indexOf(item.serverType) !== -1;
+    });
+    console.log(this.userService, 'this.userService')
```
Comment (narek118) [inline src/app/servers/server-list/server-list.component.ts]: remove

**Review by narek118 — CHANGES_REQUESTED** [review]

File: src/app/register/register.component.ts (hunk 5 above)
Comment (gorg16) [reply src/app/register/register.component.ts]: ok

File: src/app/servers/server-list/server-list.component.ts (hunk 12 above)
Comment (gorg16) [reply src/app/servers/server-list/server-list.component.ts:23]: should or must?

**Review by narek118 — APPROVED** [review]

**Pull request merged**